docs/guides/workloads/overview/index.md

```diff
@@ -12,76 +12,84 @@ menu_name: docs_{{ .version }}
 
-This guide will show you how Stash backs up and restores volumes of various workload types (Deployment, StatefulSet, DaemonSet etc.).
+This guide will show you how KubeStash backs up and restores volumes of various workload types (Deployment, StatefulSet, DaemonSet etc.).
 
 ## Before You Begin
 
-- You should be familiar with the following `Stash` concepts:
+- You should be familiar with the following `KubeStash` concepts:
   - [BackupConfiguration](/docs/concepts/crds/backupconfiguration/index.md)
   - [BackupSession](/docs/concepts/crds/backupsession/index.md)
   - [RestoreSession](/docs/concepts/crds/restoresession/index.md)
-  - [Repository](/docs/concepts/crds/repository/index.md)
+  - [BackupStorage](/docs/concepts/crds/backupstorage/index.md)
 
 ## How Backup Process Works
 
-The following diagram shows how Stash takes backup of the volumes of a workload. Open the image in a new tab to see the enlarged version.
+The following diagram shows how KubeStash takes backup of the volumes of a workload. Open the image in a new tab to see the enlarged version.
 
 <figure align="center">
-  <img alt="Stash Backup Flow" src="images/backup_overview.svg">
-<figcaption align="center">Fig: Backup process of Workload volumes in Stash</figcaption>
+   <img alt="KubeStash Backup Flow" src="images/backup_overview.svg">
+    <figcaption align="center">Fig: Backup process of Workload volumes in KubeStash</figcaption>
 </figure>
 
 The backup process consists of the following steps:
 
-1. At first, a user creates a Secret. This secret holds the credentials to access the backend where the backed up data will be stored.
+1. At first, a user creates a `Secret`. This secret holds the credentials to access the backend where the backed up data will be stored.
 
-2. Then, she creates a `Repository` crd which represents the original repository in the backend.
+2. Then, she creates a `BackupStorage` custom resource that specifies the backend information, along with the `Secret` containing the credentials needed to access the backend.
 
-3. Then, she creates a `BackupConfiguration` crd which specifies the targeted workload and desired file paths to backup. It also specifies the `Repository` object that holds the backend information where the backed up data will be stored.
+3. KubeStash operator watches for `BackupStorage` custom resources. When it finds a `BackupStorage` object, it initializes the `BackupStorage` by uploading the `metadata.yaml` file into the target storage.
 
-4. Stash operator watches for `BackupConfiguration` objects.
+4. Then, she creates a `BackupConfiguration` custom resource which specifies the targeted workload and desired file paths to backup. The `BackupConfiguration` object specifies the `Repository` pointing to a `BackupStorage` that contains backend information, indicating where to upload backup data. It also defines the `Addon` information with a specified tasks and their configuration parameters to be used for backing up the volume.
```

Comment:
Then, she creates a `BackupConfiguration` custom resource that specifies the targeted workload, the Addon info with specified tasks, etc. It also provides information about one or more repositories, each indicating a path and a `BackupStorage` for storing the backed-up data.